### PR TITLE
change function inputs

### PR DESCRIPTION
**Pull request by yairchn**
I am changing the input vars for NetCDFIO_Diags so that in my future PR with grid search I can all 'NetCDFIO_Diags' without 'ekp' as input

### CODE REVIEW

File: src/NetCDFIO.jl [hunk 1]
```diff
@@ -30,7 +30,8 @@ mutable struct NetCDFIO_Diags
         config::Dict{Any, Any},
         outdir_path::String,
         ref_stats::ReferenceStatistics,
-        ekp::EnsembleKalmanProcess,
+        p::Int64,
+        N_ens::Int64,
         priors::ParameterDistribution,
         val_ref_stats::Union{ReferenceStatistics, Nothing} = nothing,
     )
```
Comment (ilopezgp) [inline src/NetCDFIO.jl:37]:
```suggestion
        N_ens::IT,
        priors::ParameterDistribution,
        val_ref_stats::Union{ReferenceStatistics, Nothing} = nothing,
    ) where {IT <: Integer}
```

File: src/NetCDFIO.jl [hunk 2]
```diff
@@ -54,7 +55,6 @@ mutable struct NetCDFIO_Diags
         NC.Dataset(filepath, "c") do root_grp
 
             # Fetch dimensionality
-            p, N_ens = size(get_u_final(ekp))
```
Comment (ilopezgp) [inline src/NetCDFIO.jl:57]: `p = length(get_dimensions(priors))`

File: src/Pipeline.jl [hunk 3]
```diff
@@ -704,7 +704,8 @@ function init_diagnostics(
     validation::Bool = false,
 )
     write_full_stats = get_entry(config["reference"], "write_full_stats", true)
-    diags = NetCDFIO_Diags(config, outdir_path, ref_stats, ekp, priors, val_ref_stats)
+    p, N_ens = size(get_u_final(ekp))
+    diags = NetCDFIO_Diags(config, outdir_path, ref_stats, p, N_ens, priors, val_ref_stats)
```
Comment (ilopezgp) [inline src/Pipeline.jl:708]:
```suggestion
    N_ens = size(get_u_final(ekp), 2)
    diags = NetCDFIO_Diags(config, outdir_path, ref_stats, N_ens, priors, val_ref_stats)
```

File: test/NetCDFIO/runtests.jl [hunk 4]
```diff
@@ -48,7 +48,8 @@ using EnsembleKalmanProcesses.EnsembleKalmanProcessModule
     config["reference"] = Dict()
     priors = construct_priors(config["prior"]["constraints"])
     ekp = EnsembleKalmanProcess(rand(2, 10), ref_stats.y, ref_stats.Γ, Inversion())
-    diags = NetCDFIO_Diags(config, data_dir, ref_stats, ekp, priors)
+    p, N_ens = size(get_u_final(ekp))
+    diags = NetCDFIO_Diags(config, data_dir, ref_stats, p, N_ens, priors)
```
Comment (ilopezgp) [inline test/NetCDFIO/runtests.jl:52]:
```suggestion
    N_ens = size(get_u_final(ekp), 2)
    diags = NetCDFIO_Diags(config, data_dir, ref_stats, N_ens, priors)
```

File: src/NetCDFIO.jl (hunk 2 above)
Comment (ilopezgp) [reply src/NetCDFIO.jl:57]: Actually, easier to do `p = get_total_dimension(priors)`

**Review by ilopezgp — APPROVED** [review]
LGTM

**Comment by yairchn** [comment]
Bors r+

**Pull request merged**